M2/Macaulay2/m2/matrix2.m2

```diff
@@ -325,32 +325,40 @@ quotientRemainder(Matrix,Matrix) := Matrix => (f,g) -> (
 	  map(M, L, rem)
      ))
 
-leftQuotientWarn = true
-
-Matrix // Matrix := Matrix => (f,g) -> quotient(f,g)
-Matrix \\ Matrix := (g,f) -> (if leftQuotientWarn then (leftQuotientWarn = false; printerr "Warning: 'm \\\\ n' is deprecated; use 'n // m' instead."); f // g)
-quotient'(Matrix,Matrix) := Matrix => (f,g) -> (
-     if not isFreeModule source f or not isFreeModule target f
-     or not isFreeModule source g or not isFreeModule target g then error "expected maps between free modules";
-     dual quotient(dual f, dual g))
-quotient(Matrix,Matrix) := Matrix => opts -> (f,g) -> (
-     if target f != target g then error "quotient: expected maps with the same target";
-     c := runHooks((quotient, Matrix, Matrix), (opts, f, g));
-     if c =!= null then c else error "quotient: no method implemented for this type of input")
-
-addHook((quotient, Matrix, Matrix), Strategy => Default, (opts, f, g) -> (
+--------------------------------------------------------------------------------
+-- factoring of a matrix through another
+
+-- Note: when f and g are endomorphisms, the sources and targets all agree,
+-- so we need both functions quotient and quotient' to distinguish them.
+
+-- factor matrices with same targets
+Matrix // Matrix := Matrix => (f, g) -> quotient(f, g)
+Number // Matrix := RingElement // Matrix := Matrix => (r, g) -> map(source g, target g, map(target g, cover target g, r) // g)
+Matrix // Number := Matrix // RingElement := Matrix => (f, r) -> map(target f, source f, f // map(target f, cover target f, r))
+
+-- factor matrices with same sources
+Matrix \\ Matrix := Matrix => (g, f) -> quotient'(f, g)
+-- commented because they don't seem very meaningful
+--Matrix \\ Number := Matrix \\ RingElement := Matrix => (g, r) -> map(source g, target g, g \\ map(cover source g, source g, r))
+--Number \\ Matrix := RingElement \\ Matrix := Matrix => (r, f) -> map(target g, source g, map(cover source g, source g, r) \\ f)
```

Comment:
Can we make the (breaking?) changes to quotients of ``Matrices`` a separate pull request?  I was unable to find the associated documentation and tests.

Reply:
I tried to do this and failed. I don't think it's impossible, but I decided it wasn't worth the time and effort to essentially redo everything in a different order.